metaflow/datatools/s3.py

```diff
@@ -873,11 +888,10 @@ def _one_boto_op(self, op, url):
             tmp = NamedTemporaryFile(dir=self._tmpdir,
                                      prefix='metaflow.s3.one_file.',
                                      delete=False)
-            self.reset_client()
-            try:    
-                op(self._s3_client, tmp.name)
+            try:
+                op(self._s3_client.client, tmp.name)
                 return tmp.name
-            except self._s3_client_error as err:
+            except self._s3_client.error as err:
```

Comment:
it is unusual to have side-effect hidden inside an `except` clause like in this case (`.error` can trigger `get_aws_client`). We could use `_s3_client_error` as before

Reply:
It no longer exists. In this case though, there is no way for the side effect to happen since the try statement accesses `.client` which would do any triggering at that point.